File: lib/layer2_core/light/e_secondary_fixed_light.dart
import '../../layer1_raw/design_token.dart';
import '../../layer1_raw/light/system.dart';
import '../../layer1_raw/light/standard.dart';
import '../../layer1_raw/light/blue.dart';
import '../../layer1_raw/light/green.dart';
import '../../layer1_raw/light/red.dart';
import '../../layer1_raw/light/nothing.dart';

class ESecondaryFixedLight {
  static DesignToken getToken(dynamic activeTheme) {
    switch (activeTheme.toString().split('.').last) {
      case 'system': return csyl29;
      case 'standard': return cstl29;
      case 'blue': return cbl29;
      case 'green': return cgl29;
      case 'red': return crl29;
      case 'nothing': return cnol29;
      default: return cstl29;
    }
  }
}

File: android/src/main/kotlin/com/example/design_engine/layer2_core/light/ESecondaryFixedLight.kt
package com.example.design_engine.layer2_core.light
import com.example.design_engine.R
import com.example.design_engine.layer3_logic.AppTheme
object ESecondaryFixedLight {
    fun getResId(activeTheme: AppTheme): Int {
        return when (activeTheme) {
            AppTheme.SYSTEM -> R.string.csyl29
            AppTheme.STANDARD -> R.color.cstl29
            AppTheme.BLUE -> R.color.cbl29
            AppTheme.GREEN -> R.color.cgl29
            AppTheme.RED -> R.color.crl29
            AppTheme.NOTHING -> R.color.cnol29
        }
    }
}
